### PR TITLE
Lazy list for even lazier acceptedTypes parsing

### PR DESCRIPTION
**Pull request by danielkec**
Fixes https://gitlab-odx.oracledx.com/helidon/helidon/-/issues/14

Signed-off-by: Daniel Kec <daniel.kec@oracle.com>

### CODE REVIEW

File: media/common/src/main/java/io/helidon/media/common/MessageBodyWriterContext.java [hunk 1]
```diff
@@ -118,7 +117,7 @@ private MessageBodyWriterContext(MessageBodyWriterContext writerContext, Paramet
         this.headers = headers;
         this.writers = new MessageBodyOperators<>(writerContext.writers);
         this.swriters = new MessageBodyOperators<>(writerContext.swriters);
-        this.acceptedTypes = writerContext.acceptedTypes;
+        this.acceptedTypes = List.copyOf(writerContext.acceptedTypes);
```
Comment (tomas-langer) [inline media/common/src/main/java/io/helidon/media/common/MessageBodyWriterContext.java:120]: As far as I can see, `List.copyOf` actually gets all the elements in the list. So the lazyness would be gone in this case.

Comment (danielkec) [reply media/common/src/main/java/io/helidon/media/common/MessageBodyWriterContext.java:120]: I am not sure if we do need the copy, it's  unmodifiable list all the time, but I could have missed something WDYT @romain-grecourt ?

Comment (romain-grecourt) [reply media/common/src/main/java/io/helidon/media/common/MessageBodyWriterContext.java:120]: Right, this list shouldn't need to be copied.

Comment (danielkec) [reply media/common/src/main/java/io/helidon/media/common/MessageBodyWriterContext.java]: fixed

**Pull request merged**